controllers/workflow_controller.go

```diff
@@ -179,25 +183,43 @@ func (r *WorkflowReconciler) Reconcile(ctx context.Context, req ctrl.Request) (r
 	workflow.Status.Status = dwsv1alpha2.StatusCompleted
 	workflow.Status.Message = ""
 
-	// Loop through the driver status array and update the workflow
-	// status as necessary
+	// Loop through the driver status array find the entries that are for the current state
+	drivers := []dwsv1alpha2.WorkflowDriverStatus{}
+
 	for _, driver := range workflow.Status.Drivers {
 		if driver.WatchState != workflow.Status.State {
 			continue
 		}
 
-		if driver.Completed == false {
-			workflow.Status.Ready = false
-			workflow.Status.Status = dwsv1alpha2.StatusDriverWait
-		}
+		drivers = append(drivers, driver)
+	}
 
-		if driver.Message != "" {
-			workflow.Status.Message = fmt.Sprintf("DW Directive %d: %s", driver.DWDIndex, driver.Message)
-		}
+	if len(drivers) > 0 {
+		// Sort the driver entries by their status
+		sort.Slice(drivers, func(i, j int) bool {
+			return statusPriority(drivers[i].Status) > statusPriority(drivers[j].Status)
```

Comment:
Is this sorting my alphabetical order? If so, how does that equal the "highest priority" in the next block?

Reply:
This sorts the drivers by the int value that statusPriority() returns. The idea here is to find the most interesting driver statuses and roll up that info into the workflow status section. So StatusError is more interesting than StatusStalled, StatusStalled is more interesting than StatusDriverWait, etc.